src/daemon/bitcoind/interface.rs

```diff
@@ -87,6 +87,57 @@ impl BitcoinD {
         self.make_node_request("getblockchaininfo", &[])
     }
 
+    pub fn get_tip(&self) -> Result<(u32, BlockHash), BitcoindError> {
+        let json_height = self.make_node_request("getblockcount", &[])?;
+        let height = json_height.as_u64().ok_or_else(|| {
+            BitcoindError("API break, 'getblockcount' didn't return an u64.".to_string())
+        })?;
+        let hash = BlockHash::from_str(
+            self.make_node_request("getblockhash", &[json_height])?
+                .as_str()
+                .ok_or_else(|| {
+                    BitcoindError("API break, 'getblockhash' didn't return a string.".to_string())
+                })?,
+        )
+        .map_err(|e| BitcoindError(format!("Invalid blockhash given by 'getblockhash': {}", e)))?;
+
+        Ok((height as u32, hash))
+    }
+
+    pub fn synchronization_info(&self) -> Result<(u64, u64, bool, f64), BitcoindError> {
```

Comment:
Maybe clearer to return a struct instead of  (u64, u64, bool, f64)

Reply:
Right

Reply:
Addressed, here is the patch i added to a6cfc9f :
```patch
commit e38a12fd1188cbc9f602dd06839da01cf9e2fdae
Author: Antoine Poinsot <darosior@protonmail.com>
Date:   Sun Dec 6 19:34:41 2020 +0100

    fixup! jsonrpc: implement the 'getinfo' call
    
    Signed-off-by: Antoine Poinsot <darosior@protonmail.com>

diff --git a/src/daemon/bitcoind/actions.rs b/src/daemon/bitcoind/actions.rs
index c130309..49a7951 100644
--- a/src/daemon/bitcoind/actions.rs
+++ b/src/daemon/bitcoind/actions.rs
@@ -1,5 +1,8 @@
 use crate::{
-    bitcoind::{interface::BitcoinD, BitcoindError},
+    bitcoind::{
+        interface::{BitcoinD, SyncInfo},
+        BitcoindError,
+    },
     database::{
         actions::{db_insert_new_vault, db_unvault_deposit, db_update_tip},
         interface::db_wallet,
@@ -71,7 +74,12 @@ fn bitcoind_sync_status(
 ) -> Result<(), BitcoindError> {
     let first_poll = sleep_duration.is_none();
 
-    let (headers, blocks, ibd, progress) = bitcoind.synchronization_info()?;
+    let SyncInfo {
+        headers,
+        blocks,
+        ibd,
+        progress,
+    } = bitcoind.synchronization_info()?;
     *sync_progress = progress;
 
     if first_poll {
diff --git a/src/daemon/bitcoind/interface.rs b/src/daemon/bitcoind/interface.rs
index cc0d8b8..c98dcdf 100644
--- a/src/daemon/bitcoind/interface.rs
+++ b/src/daemon/bitcoind/interface.rs
@@ -104,22 +104,22 @@ impl BitcoinD {
         Ok((height as u32, hash))
     }
 
-    pub fn synchronization_info(&self) -> Result<(u64, u64, bool, f64), BitcoindError> {
+    pub fn synchronization_info(&self) -> Result<SyncInfo, BitcoindError> {
         let chaininfo = self.make_node_request("getblockchaininfo", &[])?;
-        Ok((
-            chaininfo
+        Ok(SyncInfo {
+            headers: chaininfo
                 .get("headers")
                 .and_then(|h| h.as_u64())
                 .ok_or_else(|| {
                     BitcoindError("No valid 'headers' in getblockchaininfo response?".to_owned())
                 })?,
-            chaininfo
+            blocks: chaininfo
                 .get("blocks")
                 .and_then(|b| b.as_u64())
                 .ok_or_else(|| {
                     BitcoindError("No valid 'blocks' in getblockchaininfo response?".to_owned())
                 })?,
-            chaininfo
+            ibd: chaininfo
                 .get("initialblockdownload")
                 .and_then(|i| i.as_bool())
                 .ok_or_else(|| {
@@ -127,7 +127,7 @@ impl BitcoinD {
                         "No valid 'initialblockdownload' in getblockchaininfo response?".to_owned(),
                     )
                 })?,
-            chaininfo
+            progress: chaininfo
                 .get("verificationprogress")
                 .and_then(|i| i.as_f64())
                 .ok_or_else(|| {
@@ -135,7 +135,7 @@ impl BitcoinD {
                         "No valid 'initialblockdownload' in getblockchaininfo response?".to_owned(),
                     )
                 })?,
-        ))
+        })
     }
 
     pub fn createwallet_startup(&self, wallet_path: String) -> Result<(), BitcoindError> {
@@ -564,3 +564,10 @@ impl BitcoinD {
         Ok(None)
     }
 }
+
+pub struct SyncInfo {
+    pub headers: u64,
+    pub blocks: u64,
+    pub ibd: bool,
+    pub progress: f64,
+}
```